eox_nelp/pearson_vue/pipeline.py

```diff
@@ -224,30 +242,40 @@ def import_exam_authorization(ead_request, **kwargs):  # pylint: disable=unused-
     Raises:
         Exception: If the Pearson VUE RTI service does not accept the import request.
     """
-    api_client = PearsonRTIApiClient()
-    payload = {
-        "soapenv:Envelope": {
-            "soapenv:Header": {
-                "wsse:Security": {
-                    "wsse:UsernameToken": {
-                        "wsse:Username": getattr(settings, "PEARSON_RTI_WSDL_USERNAME"),
-                        "wsse:Password": {
-                            "#text": getattr(settings, "PEARSON_RTI_WSDL_PASSWORD")
+    @audit_method(action="Import Exam Authorization")
+    def import_exam_authorization_request(ead_request):
+        """This is a wrapper that allows to make audit-able the import_exam_authorization method."""
+        api_client = PearsonRTIApiClient()
+        payload = {
+            "soapenv:Envelope": {
+                "soapenv:Header": {
+                    "wsse:Security": {
+                        "wsse:UsernameToken": {
+                            "wsse:Username": getattr(settings, "PEARSON_RTI_WSDL_USERNAME"),
+                            "wsse:Password": {
+                                "#text": getattr(settings, "PEARSON_RTI_WSDL_PASSWORD")
+                            },
                         },
                     },
                 },
+                "soapenv:Body": {
+                    "sch:eadRequest": ead_request
+                },
             },
-            "soapenv:Body": {
-                "sch:eadRequest": ead_request
-            },
-        },
-    }
-    payload = update_xml_with_dict(PAYLOAD_EAD, payload)
-    response = api_client.import_exam_authorization(payload)
+        }
+        payload = update_xml_with_dict(PAYLOAD_EAD, payload)
+        response = api_client.import_exam_authorization(payload)
+
+        if response.get("status", "error") == "accepted":
+            return response
 
-    if response.get("status", "error") != "accepted":
+        logger.info("Import exam authorization pipeline has failed with the following response: %s", response)
         # pylint: disable=broad-exception-raised
-        raise Exception("Error trying to process import exam authorization request.")
+        raise Exception(
```

Comment:
I like that this Exception gives the _**Fail**_ status, but what do you think of the retry of the error like 4 times because it is en celery async environment.
I think that it would retry 4 times, saving the 4 Fail record. The good news is that now it is easier to distinguish than the **_success_** records.

Reply:
Short answer out of the scope, long answer I was thinking to implement a PearsonBaseError here and limit the retrying process just to the ping pipeline since other case will need that a user o someone changes something so it's not necessary multiple attempts

Reply:
I really like that comment.